homeassistant/components/flux.py

```diff
@@ -0,0 +1,228 @@
+    vol.Optional(CONF_AUTO, default=False): vol.Coerce(bool),
+    vol.Optional(CONF_DAY_CT, default=4000): vol.Coerce(int),
+    vol.Optional(CONF_SUNSET_CT, default=3000): vol.Coerce(int),
+    vol.Optional(CONF_BEDTIME_CT, default=1900): vol.Coerce(int),
+    vol.Optional(CONF_BRIGHTNESS, default=90): vol.Coerce(int)
+})
+
+
+def set_lights_xy(hass, lights, x_value, y_value, brightness):
+    """Set color of array of lights."""
+    for light in lights:
+        if is_on(hass, light):
+            turn_on(hass, light,
+                    xy_color=[x_value, y_value],
+                    brightness=brightness)
+
+
+# https://github.com/KpaBap/hue-flux/blob/master/hue-flux.py
+def rgb_to_xy(red, green, blue):
+    """Convert RGB to xy."""
+    red = red / 255
+    blue = blue / 255
+    green = green / 255
+
+    # Gamma correction
+    red = pow((red + 0.055) / (1.0 + 0.055),
+              2.4) if (red > 0.04045) else (red / 12.92)
+    green = pow((green + 0.055) / (1.0 + 0.055),
+                2.4) if (green > 0.04045) else (green / 12.92)
+    blue = pow((blue + 0.055) / (1.0 + 0.055),
+               2.4) if (blue > 0.04045) else (blue / 12.92)
+
+    tmp_x = red * 0.664511 + green * 0.154324 + blue * 0.162028
+    tmp_y = red * 0.313881 + green * 0.668433 + blue * 0.047685
+    tmp_z = red * 0.000088 + green * 0.072310 + blue * 0.986039
+
+    try:
+        x_value = tmp_x / (tmp_x + tmp_y + tmp_z)
+        y_value = tmp_y / (tmp_x + tmp_y + tmp_z)
+    except ZeroDivisionError:
+        x_value, y_value = 0, 0
+
+    if x_value > 0.675:
+        x_value = 0.675
+    if x_value < 0.167:
+        x_value = 0.167
+
+    if y_value > 0.518:
+        y_value = 0.518
+    if y_value < 0.04:
+        y_value = 0.04
+
+    return round(x_value, 3), round(y_value, 3)
+
+
+# https://github.com/KpaBap/hue-flux/blob/master/hue-flux.py
+def colortemp_k_to_rgb(temp):
+    """Convert temp to RGB."""
+    temp = temp / 100
+
+    if temp <= 66:
+        red = 255
+    else:
+        red = temp - 60
+        red = 329.698727446 * (red ** -0.1332047592)
+        red = 0 if red < 0 else red
+        red = 255 if red > 255 else red
+
+    if temp <= 66:
+        green = temp
+        green = 99.4708025861 * math.log(green) - 161.1195681661
+        green = 0 if green < 0 else green
+        green = 255 if green > 255 else green
+    else:
+        green = temp - 60
+        green = 288.1221695283 * (green ** -0.0755148492)
+        green = 0 if green < 0 else green
+        green = 255 if green > 255 else green
+
+    if temp >= 66:
+        blue = 255
+    else:
+        if temp <= 19:
+            blue = 0
+        else:
+            blue = temp - 10
+            blue = 138.5177312231 * math.log(blue) - 305.0447927307
+            blue = 0 if blue < 0 else blue
+            blue = 255 if blue > 255 else blue
+
+    return red, green, blue
+
+
+def setup(hass, config):
+    """Flux setup."""
+    flux = Flux(hass, config)
```

Comment:
if you do `config.get(DOMAIN)` here you can leave that from the code below. Also, since you defined a `PLATFORM_SCHEMA` your config got converted to a list, since you'll rename that you can drop the `[0]`